src/pages/tBTC/Bridge/Minting/Timeline.tsx

```diff
@@ -0,0 +1,196 @@
+      </>
+    ),
+  },
+  {
+    id: MintingStep.Deposit,
+    label: "Make a BTC deposit",
+    description: (
+      <>
+        Send any amount lager than 0.01 BTC to this unique BTC Deposit Address.
+        The amount sent will be used to mint tBTC.
+        <StyledBadge>ACTION ON BITCOIN</StyledBadge>
+      </>
+    ),
+  },
+  {
+    id: MintingStep.InitiateMinting,
+    label: "Initiate minting",
+    description: (
+      <>
+        Minting tBTC does not require you to wait for the Bitcoin confirmations.
+        Sign an Ethereum transaction in your wallet and your tBTC will arrive in
+        around 1 to 3 hours.
+        <StyledBadge>ACTION ON ETHEREUM</StyledBadge>
+      </>
+    ),
+  },
+]
+
+const depositDetailsItems: TimelineItemType[] = [
+  {
+    id: DepositDetailsStep.BitcoinConfirmations,
+    label: "Bitcoin Confirmations",
+    description:
+      "Minting tBTC does not require you to wait for the Bitcoin confirmations. Sign an Ethereum transaction in your wallet and your tBTC will arrive in around 1 to 3 hours.",
+  },
+  {
+    id: DepositDetailsStep.MintingInitialized,
+    label: "Minter Check",
+    description: (
+      <>
+        Our Minters are inspecting the deposit. A Minter will initiate the token
+        issuance if the deposit is confirmed.
+        <br />
+        <br />A Minter is a software empowered by the Threshold DAO. They
+        provide security against malicious deposits.
+      </>
+    ),
+  },
+  {
+    id: DepositDetailsStep.GuardianCheck,
+    label: "Guardian Check",
+    description: (
+      <>
+        Our Guardians are inspecting the deposit. This is the second layer of
+        deposit validation.
+        <br />
+        <br />
+        Guardians are software empowered by the DAO. They can deny token
+        issuance, if the Bitcoin deposit is not confirmed.
+      </>
+    ),
+  },
+  {
+    id: DepositDetailsStep.Completed,
+    label: "Completing Deposit",
+    description: "The contract is sending and minting your tBTC tokens.",
+  },
+]
+
+type MintingTimelineProps = {
+  title: string
+} & BoxProps
+
+export const Timeline: FC<MintingTimelineProps> = ({ title, ...restProps }) => {
+  const { mintingStep, depositStep } = useTbtcState()
+  const { account } = useWeb3React()
+  const { pathname } = useLocation()
+  const isDepositPageRouteActive = pathname.startsWith("/tBTC/mint/deposit")
+
+  const currentIndex = useMemo(() => {
+    const items = Object.values(
+      isDepositPageRouteActive ? DepositDetailsStep : MintingStep
+    )
+    const step = isDepositPageRouteActive ? depositStep : mintingStep
+    // if wallet is connected the state is controlled programatically
+    // based on the minting step, otherwise the state is controlled by the
+    // user
+    return account || isDepositPageRouteActive ? items.indexOf(step) : undefined
+  }, [mintingStep, depositStep, account, isDepositPageRouteActive])
+
+  const items = useMemo<TimelineItemProps[]>(() => {
```

Comment:
Non-blocking:

Would be nice to have a comment that describes what happends in this `useMemo`